### PR TITLE
bpo-38662: Invoke pip via runpy, in ensurepip

### PR DESCRIPTION
**Pull request by pradyunsg**
From an inspired idea in https://github.com/python/cpython/pull/16782#discussion_r335285656

https://bugs.python.org/issue38662

### CODE REVIEW

**Comment by pradyunsg** [comment]
Should this get a skip-news label?

File: Lib/ensurepip/__init__.py [hunk 1]
```diff
@@ -26,9 +27,17 @@ def _run_pip(args, additional_paths=None):
...
+    backup_argv = sys.argv[:]
+    sys.argv[1:] = args
+    try:
+        runpy.run_module("pip", run_name="__main__", alter_sys=True)
+    except SystemExit as e:
+        return e.code
+    finally:
+        sys.argv[:] = backup_argv
+
+    assert 0, "should never reach here"
```
Comment (eamanu) [inline Lib/ensurepip/__init__.py:40]: Why don't just write a return o raise an exception instead of the message "Should never reach here"?

Comment (pradyunsg) [reply Lib/ensurepip/__init__.py:40]: Well, they'd be functionally the same as far as I'm concerned. This is just a sanity-check, so I'm using an assert.

Anyway, if this is *the problem* with this PR, I'll be happy to change it.

Comment (ncoghlan) [reply Lib/ensurepip/__init__.py:40]: "raise SystemError" complaining that pip didn't exit as expected would be the way to go here.

**Review by ncoghlan — CHANGES_REQUESTED** [review]
I was skeptical, but the draft PR has convinced me that this is a good idea.

Changes needed are the discussed SystemError, and adding a news entry with blurb.

**Comment by csabella** [comment]
@pradyunsg, please address the review comments.  Thanks!

**Comment by hroncok** [comment]
I will submit a new PR with the requested changes. We agreed on this with @pradyunsg on IRC.

**Comment by hroncok** [comment]
In here: https://github.com/python/cpython/pull/18901

**Comment by pradyunsg** [comment]
Closing in favor of #18901

**Pull request closed without merging**